tools/distpartitioning/dataset_utils.py

```diff
@@ -33,46 +33,63 @@ def get_dataset(input_dir, graph_name, rank, num_node_weights):
     """
     #node features dictionary
     node_features = {}
+    node_feature_tids = {}
 
-    #iterate over the sub-dirs and extract the nodetypes
-    #in each nodetype folder read all the features assigned to 
-    #current rank
-    siblings = os.listdir(input_dir)
-    for s in siblings:
-        if s.startswith("nodes-"):
-            tokens = s.split("-")
-            ntype = tokens[1]
-            num_feats = tokens[2]
-            for idx in range(int(num_feats)):
-                feat_file = s +'/node-feat-'+'{:02d}'.format(idx) +'/'+ str(rank)+'.npy'
-                if (os.path.exists(input_dir+'/'+feat_file)):
-                    features = np.load(input_dir+'/'+feat_file)
-                    node_features[ntype+'/feat'] = torch.tensor(features)
+    #iterate over the "node_data" dictionary in the schema_map
+    #read the node features if exists
+    #also keep track of the type_nids for which the node_features are read.
+    dataset_features = schema_map["node_data"]
+    for ntype_name, ntype_feature_data in dataset_features.items():
+        #ntype_feature_data is a dictionary
+        #where key: feature_name, value: list of lists
+        node_feature_tids[ntype_name] = []
+        for feat_name, feat_data in ntype_feature_data.items():
+            assert len(feat_data) == world_size
+            my_feat_data = feat_data[rank]
+            if (os.path.isabs(my_feat_data[0])):
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(my_feat_data[0]))
+            else:
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(input_dir+my_feat_data[0]))
 
-    #done build node_features locally. 
-    if len(node_features) <= 0: 
-        print('[Rank: ', rank, '] This dataset does not have any node features')
-    else: 
-        for k, v in node_features.items():
-            print('[Rank: ', rank, '] node feature name: ', k, ', feature data shape: ', v.size())
+            v = [feat_name, my_feat_data[1], my_feat_data[2]]
+            l = node_feature_tids[ntype_name]
+            l.append(v)
+            node_feature_tids[ntype_name] = l
```

Comment:
the code above is very weird. how is it different from
```
node_feature_tids[ntype_name].append([feat_name, my_feat_data[1], my_feat_data[2]])
```

Reply:
Agreed. The suggested code is exactly the same as the piece of code pointed out in this CI comment. Changed the code accordingly.